app/forms/floating-ip-create.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, you can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * Copyright Oxide Computer Company
+ */
+import { useMemo } from 'react'
+import { useWatch } from 'react-hook-form'
+import { useNavigate } from 'react-router-dom'
+
+import {
+  apiQueryClient,
+  useApiMutation,
+  useApiQueryClient,
+  usePrefetchedApiQuery,
+  type FloatingIpCreate,
+  type SiloIpPool,
+} from '@oxide/api'
+import { validateIp } from '@oxide/util'
+
+import {
+  DescriptionField,
+  ListboxField,
+  NameField,
+  SideModalForm,
+  TextField,
+} from 'app/components/form'
+import { useForm, useProjectSelector, useToast } from 'app/hooks'
+import { pb } from 'app/util/path-builder'
+
+CreateFloatingIpSideModalForm.loader = async () => {
+  await apiQueryClient.prefetchQuery('projectIpPoolList', {
+    query: { limit: 1000 },
+  })
+  return null
+}
+
+export function CreateFloatingIpSideModalForm() {
+  // Fetch 1000 to we can be sure to get them all.
+  const { data: allPools } = usePrefetchedApiQuery('projectIpPoolList', {
+    query: { limit: 1000 },
+  })
+
+  const defaultPool = useMemo(
+    () => allPools.items.find((p) => p.isDefault)?.name,
+    [allPools]
+  )
+
+  const defaultValues: FloatingIpCreate = {
+    name: '',
+    description: '',
+    pool: defaultPool,
+    address: undefined,
+  }
+
+  const queryClient = useApiQueryClient()
+  const projectSelector = useProjectSelector()
+  const addToast = useToast()
+  const navigate = useNavigate()
+
+  const createFloatingIp = useApiMutation('floatingIpCreate', {
+    onSuccess() {
+      queryClient.invalidateQueries('floatingIpList')
+      addToast({ content: 'Your Floating IP has been created' })
+      navigate(pb.floatingIps(projectSelector))
+    },
+  })
+
+  const form = useForm({ defaultValues })
+
+  const toListboxItem = (p: SiloIpPool) => {
```

Comment:
If you add `defaultPool` as an argument, this function could be static, defined at top level outside of the component.